sql/core/src/main/scala/org/apache/spark/sql/execution/joins/BroadcastHashJoinExec.scala

```diff
@@ -223,19 +223,19 @@ case class BroadcastHashJoinExec(
    * Handles NULL-aware anti join (NAAJ) separately here.
    */
   protected override def codegenAnti(ctx: CodegenContext, input: Seq[ExprCode]): String = {
-    if (isNullAwareAntiJoin) {
-      val (broadcastRelation, relationTerm) = prepareBroadcast(ctx)
+    val (broadcastRelation, relationTerm) = prepareBroadcast(ctx)
+    val numOutput = metricTerm(ctx, "numOutputRows")
+    if (broadcastRelation.value == EmptyHashedRelation) {
+      s"""
+         |// If the right side is empty, AntiJoin simply returns the left side.
+         |$numOutput.add(1);
+         |${consume(ctx, input)}
+       """.stripMargin
+    } else if (isNullAwareAntiJoin) {
       val (keyEv, anyNull) = genStreamSideJoinKey(ctx, input)
       val (matched, _, _) = getJoinCondition(ctx, input)
-      val numOutput = metricTerm(ctx, "numOutputRows")
 
-      if (broadcastRelation.value == EmptyHashedRelation) {
-        s"""
-           |// If the right side is empty, NAAJ simply returns the left side.
-           |$numOutput.add(1);
-           |${consume(ctx, input)}
-         """.stripMargin
-      } else if (broadcastRelation.value == EmptyHashedRelationWithAllNullKeys) {
+      if (broadcastRelation.value == EmptyHashedRelationWithAllNullKeys) {
```

Comment:
shall we handle this case in AQE as well?

Reply:
I think EmptyHashedRelationWithAllNullKeys should only be handled in NAAJ scenario. It might be needed to consider in my next follow up => support NAAJ in ShuffleHashJoinExec.

Reply:
do you mean to eliminate NAAJ BroadcastJoin if it is a EmptyHashedRelationWithAllNullKeys?